File: Trabalho Prático JS/model/ContaCorrente.js
import { Conta } from "./Conta.js";
import { ExcecaoChequeEspecial } from "../errors/ExcecaoChequeEspecial.js";

export class ContaCorrente extends Conta {
    #chequeEspecial;

    constructor(saldoInicial, numeroConta, chequeEspecial) {
        super(saldoInicial, numeroConta);
        this.#chequeEspecial = chequeEspecial;
    }
    
    sacar(valor){
        if(valor <= (this.saldo + this.#chequeEspecial)) {
            this.saldo = (this.saldo - valor);
        } else if (this.#chequeEspecial == 0) {
            throw new ExcecaoChequeEspecial("Não há cheque especial.", valor - this.saldo);
        } else {
            throw new ExcecaoChequeEspecial("Cheque especial insuficiente.", valor - this.saldo);
        }
    }
}

File: exercicio13/Banco3/src/main/java/banco/ContaCorrente.java
package banco;

public class ContaCorrente extends Conta{
    private double chequeEspecial;

    public ContaCorrente(double saldoInicial ,double chequeEspecial) {
        super(saldoInicial);
        this.chequeEspecial = chequeEspecial;
    }

    public ContaCorrente(double saldoInicial) {
        super(saldoInicial);

    }

    @Override
    public boolean sacar(double valor){
        if(valor< saldo + chequeEspecial){
            if ((saldo-valor) > (-1*chequeEspecial)){
                saldo = saldo - valor;
                return true;
            }
            return false;
        } else if (valor > saldo + chequeEspecial) {
            return false;
        }
        return false;
    }
}
